### PR TITLE
fix(Script/Quest): Arelion's Secret. Add combat sequence to Magister …

### PR DESCRIPTION
**Pull request by LeFroid**
…Aledis

Co-authored-by: SnapperRy

Taken/modified from original TC commit in 2016: https://github.com/TrinityCore/TrinityCore/commit/94c41c92b35e20c47462113ad88b3d382dd9c44d

## Changes Proposed:
-  Adds combat script to Magister Aledis (proper behavior)
-  

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/14439 

## SOURCE:

https://youtu.be/nEm2H3OenQ0?t=66
https://www.wowhead.com/wotlk/quest=10286/arelions-secret
https://www.wowhead.com/wotlk/npc=20159/magister-aledis#ends

## Tests Performed:

- Followed following steps to test the behavior.
- .q add 10286
- .go c id 20159
- Confirm player is unable to turn in quest
- Talk to NPC, select dialog option which leads to combat after brief conversation
- At 20% HP, Aledis will stop combat and the player will be able to hand in the quest.

## How to Test the Changes:

1. .q add 10286
2. .go c id 20159
3. Talk to NPC, engage in combat
4. At 20% HP, Aledis will stop combat and the player will be able to hand in the quest.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Kitzunu** [comment]
This could most likely be done in Smart AI

**Comment by LeFroid** [comment]
I am not sure if it’s worth the effort to do a rewrite and additional testing, unless anyone would like help.

>

**Comment by FrancescoBorzi** [comment]
what's the status of this PR?

**Comment by LeFroid** [comment]
It’s working as intended as it stands. Better than the current behavior which is unscripted.On Jun 29, 2023, at 10:51 AM, Francesco Borzì ***@***.***> wrote:﻿
what's the status of this PR?

**Comment by FrancescoBorzi** [comment]
> It’s working as intended as it stands. Better than the current behavior which is unscripted edit: merge conflicts resolved also

thanks, can anyone please check this? @azerothcore/testers @azerothcore/developers

File: src/server/scripts/Outland/zone_hellfire_peninsula.cpp [hunk 1]
```diff
@@ -390,6 +390,124 @@ class npc_fel_guard_hound : public CreatureScript
...
+                    DoCast(SPELL_FIREBALL);
+                    _events.ScheduleEvent(EVENT_FIREBALL, 10s);
+                    break;
+                case EVENT_FROSTNOVA:
+                    DoCastAOE(SPELL_FROSTNOVA);
+                    _events.ScheduleEvent(EVENT_FROSTNOVA, 20s);
+                    break;
+                case EVENT_EVADE:
+                    EnterEvadeMode();
+                    break;
```
Comment (Nefertumm) [inline src/server/scripts/Outland/zone_hellfire_peninsula.cpp:483]: Indentation.

File: src/server/scripts/Outland/zone_hellfire_peninsula.cpp [hunk 2]
```diff
@@ -390,6 +390,124 @@ class npc_fel_guard_hound : public CreatureScript
...
+                }
+            }
+
+            if (!UpdateVictim())
+                return;
```
Comment (Nefertumm) [inline src/server/scripts/Outland/zone_hellfire_peninsula.cpp:488]: Uhm, maybe move this above the events, and if it's needed add another eventMap for non-combat events (or you could use TaskScheduler.

**Review by Nefertumm — CHANGES_REQUESTED** [review]
Move the script to the new system (the struct is only needed, you could drop the CreatureScript class and only leave the struct as it is, then to the AddSC_hellfire_peninsula, instead of doing new bla bla bla, you do RegisterCreatureScript(npc_magister_aledis) ).

File: src/server/scripts/Outland/zone_hellfire_peninsula.cpp (hunk 1 above)
Comment (LeFroid) [reply src/server/scripts/Outland/zone_hellfire_peninsula.cpp:478]: Updated

File: src/server/scripts/Outland/zone_hellfire_peninsula.cpp (hunk 2 above)
Comment (LeFroid) [reply src/server/scripts/Outland/zone_hellfire_peninsula.cpp]: Changed this slightly
```cpp
if (UpdateVictim())
{
    DoMeleeAttackIfReady();
}
```

Comment (LeFroid) [reply src/server/scripts/Outland/zone_hellfire_peninsula.cpp]: @Nefertumm do you see any issues with the latest changes? Is the rationale clear?

**Review by Nefertumm — APPROVED** [review]

File: src/server/scripts/Outland/zone_hellfire_peninsula.cpp (hunk 2 above)
Comment (Nefertumm) [reply src/server/scripts/Outland/zone_hellfire_peninsula.cpp]: If it works, then it's fine for me :)

**Pull request merged**